### ISSUE TITLE
Cloudflare compatibility and mitigation layer 7 dos/ddos

### ISSUE THREAD

**Issue by karek314** [jgmdev/ddos-deflate#27]
Since this tool is no longer developed and can't provide protection in case we use cdn like cloudflare. If anyone needs please take a look on my approach here - https://github.com/karek314/ddos-deflate-nginx-cloudflare

**Comment by jgmdev** [comment]
So I have updated the script to support ipv6 and some other user contributions like ip ranges and cidr with other improvements... I was thinking about this issue and thought that maybe it is possible to block connections by ip on servers behind cloudflare by inspecting the sent packets, and found on [stackoverflow](https://stackoverflow.com/questions/825481/iptable-rule-to-drop-packet-with-a-specific-substring-in-payload) :smile:  that one can match a string on a packet like this to be droped:
```
iptables -A INPUT -m string --algo bm --string "CF-Connecting-IP: A.B.C.D" -j DROP
```
I partially saw your fork and was thinking that maybe we can add some option to scan a log of ip connections made to a webserver like you do, but try it on a more generic way since there are people that uses apache, others hiawatha, etc... we would need to know how this logs are generated on apache, hiawatha, [insert another web server here] etc... when running behind the cloudflare stuff, and test if the above iptable rules does works by running some benchmarking tool like ab or wrk to a site behind cloudflare (if possible).

**Comment by jgmdev** [comment]
Implemented as described in #59.
